controller/instance_manager_controller.go

```diff
@@ -419,6 +425,92 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 	return nil
 }
 
+func (imc *InstanceManagerController) syncInstanceManagerPDB(im *longhorn.InstanceManager) error {
+	if im.Status.CurrentState != types.InstanceManagerStateRunning {
+		return nil
+	}
+
+	unschedulable, err := imc.isNodeUnschedulable()
+	if err != nil {
+		return err
+	}
+
+	imPDB, err := imc.ds.GetPDBRO(imc.getPDBName(im))
+	if err != nil && !datastore.ErrorIsNotFound(err) {
+		return err
+	}
+
+	// Delete the PodDisruptionBudget for the instance manager if all of the following conditions are meet:
+	// 1. The current node is unschedulable. This is an signal that the node is being cordoned/drained.
+	// 2. There is no instance process inside the instance manager
+	// 3. There exist a PodDisruptionBudget for this instance manager
+	if unschedulable && len(im.Status.Instances) == 0 {
```

Comment:
What if the user want to depend on the rebuilding on the existing replica feature to quickly reboot the node without migrating all the replicas to other nodes? Wait for replicas to finish eviction can take a long time.

The PDB always makes sense for the engine since it should be moved with the workload pod. But It really depends on the user if they want to apply it to the replicas. If they want, it's likely they will perform the replica eviction first.

Reply:
@yasker Great point!

Replica eviction feature will be handled replica migration. 
Blindly waiting for replica migration will take long time. 

I will modify the code so that Longhorn only use PDB to protect engine IM pods

Reply:
I fixed it